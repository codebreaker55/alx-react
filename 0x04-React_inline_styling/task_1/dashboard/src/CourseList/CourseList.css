table {
  margin: 0;
  width: auto;
  border: none;
  font-size: inherit;
}

th, td {
  border: none;
  width: auto;
}

tr:nth-child(2) {
  text-align: inherit;
}
