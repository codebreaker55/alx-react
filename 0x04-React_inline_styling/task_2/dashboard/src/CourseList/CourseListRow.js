import React from "react";
import PropTypes from "prop-types";
import { StyleSheet, css } from "aphrodite";

function CourseListRow({ isHeader, textFirstCell, textSecondCell }) {
  const rowStyle = isHeader ? css(styles.headerRow) : css(styles.normalRow);
  
  return (
    <tr className={rowStyle}>
      {isHeader ? (
        textSecondCell === null ? (
          <th colSpan={2} className={css(styles.headerCell)}>
            {textFirstCell}
          </th>
        ) : (
          <>
            <th className={css(styles.headerCell)}>{textFirstCell}</th>
            <th className={css(styles.headerCell)}>{textSecondCell}</th>
          </>
        )
      ) : (
        <>
          <td className={css(styles.normalCell)}>{textFirstCell}</td>
          <td className={css(styles.normalCell)}>{textSecondCell}</td>
        </>
      )}
    </tr>
  );
}

const styles = StyleSheet.create({
  headerRow: {
    backgroundColor: "#deb5b545",
  },
  normalRow: {
    backgroundColor: "#f5f5f5ab",
  },
  headerCell: {
    fontWeight: 'bold',
    textAlign: 'left',
    padding: '10px',
  },
  normalCell: {
    textAlign: 'left',
    padding: '10px',
  },
});

CourseListRow.propTypes = {
  isHeader: PropTypes.bool,
  textFirstCell: PropTypes.string.isRequired,
  textSecondCell: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

CourseListRow.defaultProps = {
  isHeader: false,
  textSecondCell: null,
};

export default CourseListRow;
