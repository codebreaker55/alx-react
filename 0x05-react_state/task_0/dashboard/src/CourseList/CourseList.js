import React from "react";
import "./CourseList.css";
import CourseListRow from "./CourseListRow";
import PropTypes from "prop-types";
import CourseShape from "./CourseShape";

function CourseList({ listCourses }) {
  return (
    <table id="CourseList">
      <thead>
        <CourseListRow textFirstCell="Course name" textSecondCell="Credit" isHeader={true} />
        <CourseListRow textFirstCell="Available courses" textSecondCell={null} isHeader={true} />
      </thead>
      <tbody>
        {listCourses.length > 0 ? (
          listCourses.map(({ id, name, credit }) => (
            <CourseListRow key={id} textFirstCell={name} textSecondCell={credit} isHeader={false} />
          ))
        ) : (
          <CourseListRow textFirstCell="No course available yet" textSecondCell={null} isHeader={false} />
        )}
      </tbody>
    </table>
  );
}

CourseList.propTypes = {
  listCourses: PropTypes.arrayOf(CourseShape),
};

CourseList.defaultProps = {
  listCourses: [],
};

export default CourseList;
