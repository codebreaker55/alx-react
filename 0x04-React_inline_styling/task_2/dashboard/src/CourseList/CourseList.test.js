import React from "react";
import CourseList from "./CourseList";
import CourseListRow from "./CourseListRow";
import { shallow } from "enzyme";
import { StyleSheetTestUtils } from "aphrodite";

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection();
});

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
});

const listCourses = [
  { id: 1, name: "ES6", credit: 60 },
  { id: 2, name: "Webpack", credit: 20 },
  { id: 3, name: "React", credit: 40 },
];

describe("CourseList component tests", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<CourseList />);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders the header row and course rows correctly", () => {
    const wrapper = shallow(<CourseList listCourses={listCourses} />);

    // check the header row
    const headerRow = wrapper.find("thead").childAt(0);
    expect(headerRow.prop("textFirstCell")).toEqual("Available courses");
    
    const headerSecondCell = headerRow.prop("textSecondCell");
    if (headerSecondCell !== null) {
      expect(headerSecondCell).toEqual("Credit");
    } else {
      expect(headerSecondCell).toBeNull();
    }

    expect(headerRow.prop("isHeader")).toBe(true);

    // check the body rows
    expect(wrapper.find("tbody").children()).toHaveLength(3);
    expect(wrapper.find("tbody").childAt(0)).toMatchSnapshot();
    expect(wrapper.find("tbody").childAt(1)).toMatchSnapshot();
    expect(wrapper.find("tbody").childAt(2)).toMatchSnapshot();
  });

  it("renders correctly when passed a list of courses", () => {
    const wrapper = shallow(<CourseList listCourses={listCourses} />);

    expect(wrapper.find("tbody").children()).toHaveLength(3);
    expect(wrapper.find("tbody").childAt(0)).toMatchSnapshot();
    expect(wrapper.find("tbody").childAt(1)).toMatchSnapshot();
    expect(wrapper.find("tbody").childAt(2)).toMatchSnapshot();
  });
});

