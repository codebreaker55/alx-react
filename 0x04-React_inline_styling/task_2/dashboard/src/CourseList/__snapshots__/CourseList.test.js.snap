// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList component tests renders correctly when passed a list of courses 1`] = `ShallowWrapper {}`;

exports[`CourseList component tests renders correctly when passed a list of courses 2`] = `ShallowWrapper {}`;

exports[`CourseList component tests renders correctly when passed a list of courses 3`] = `ShallowWrapper {}`;

exports[`CourseList component tests renders the header row and course rows correctly 1`] = `ShallowWrapper {}`;

exports[`CourseList component tests renders the header row and course rows correctly 2`] = `ShallowWrapper {}`;

exports[`CourseList component tests renders the header row and course rows correctly 3`] = `ShallowWrapper {}`;
